jcl/src/java.base/share/classes/java/lang/J9VMInternals.java

```diff
@@ -333,7 +333,9 @@ private static void threadCleanup(Thread thread) {
 	private static void checkPackageAccess(final Class clazz, ProtectionDomain pd) {
 		final SecurityManager sm = System.getSecurityManager();
 		if (sm != null) {
-			AccessController.doPrivileged(new PrivilegedAction() {
+			ProtectionDomain[] pdArray = (pd == null) ? new ProtectionDomain[]{} : new ProtectionDomain[]{pd};
```

Comment:
I don't think creation of an empty ProtectionDomain[] is required, null should be sufficient.

Reply:
There are constructors not checking `null` for incoming `ProtectionDomain` array. 
Will modify them instead.

Reply:
Actually incoming `ProtectionDomain` array can't be `null`.
https://docs.oracle.com/javase/9/docs/api/java/security/AccessControlContext.html#AccessControlContext-java.security.ProtectionDomain:A-
```
public AccessControlContext​(ProtectionDomain[] context)

Create an AccessControlContext with the given array of ProtectionDomains. Context must not be null. Duplicate domains will be removed from the context.

Parameters:
    context - the ProtectionDomains associated with this context. The non-duplicate domains are copied from the array. Subsequent changes to the array will not affect this AccessControlContext.
Throws:
    NullPointerException - if context is null 
```

Reply:
AccessControlContext​(ProtectionDomain[]) isn't being called, an internal constructor is being called and it seems to handle null for the ProtectionDomain[]. There is already code which sets AccessControlContext.context = null

Reply:
Sorry, I commented on the wrong part of the change. I meant this comment for the callers of AccessControlContext(AccessController, ProtectionDomain[], int)

Reply:
Updated PR accordingly, verified manually that `pConfig` still compiles.
Pls have another look.